src/components/tiptap/floatingMenu/FloatingMenu.tsx

```diff
@@ -0,0 +1,208 @@
+import { FloatingMenu, Editor } from '@tiptap/react'
+import { FC } from 'react';
+import { TiptapEditorUtils } from '@/utils/tiptapEditorUtils';
+import { ImagePickerUtils } from '@/utils/imagePickerUtils';
+import { useAppState } from '@/hooks/useAppState';
+import { H1Icon, H2Icon, H3Icon, TextIcon, AutofillIcon, NumberedListIcon, BulletListIcon, UploadIcon2, CalloutIcon, LinkIcon, TableIcon, EmbedIcon } from '@/icons';
+
+interface IFloatingMenuContainer {
+  editor: Editor;
+}
+
+const FloatingMenuContainer: FC<IFloatingMenuContainer> = ({ editor }) => {
+
+  const tiptapEditorUtils = new TiptapEditorUtils(editor)
+  const imagePickerUtils = new ImagePickerUtils()
+
+  const appState = useAppState()
+
+
+  return (
+    <>
+      <FloatingMenu editor={editor} tippyOptions={{ duration: 100 }}
+        shouldShow={({ view }) => {
+          //@ts-ignore
+          const { $cursor } = view.state.selection;
+
+          if ($cursor) {
+            const { pos } = $cursor
+            const charBeforeCursor = view.state.doc.textBetween(pos - 1, pos)
+
+            return charBeforeCursor === '/'
+
+          }
+          return false
+
+        }} pluginKey="optionsMenu">
+        <div className='flex flex-col gap-0.5 bg-white py-2 border border-new-card-border rounded shadow-vairant-1 absolute top-3 w-52'>
+          <button className='flex flex-row gap-x-2.5 items-center py-1.5 px-3 focus:bg-new-white-2 cursor-pointer outline-none'
+            onClick={() => {
+              tiptapEditorUtils.clearCurrentLineContent()
+              tiptapEditorUtils.toggleHeading(1)
+            }}
+          >
+            <div>
+              <H1Icon />
+            </div>
+            <div>
+              <p className='text-sm'>Heading 1</p>
+            </div>
+          </button>
+
+          <button className='flex flex-row gap-x-2.5 items-center py-1.5 px-3 focus:bg-new-white-2 cursor-pointer outline-none'
+            onClick={() => {
+              tiptapEditorUtils.clearCurrentLineContent()
+              tiptapEditorUtils.toggleHeading(2)
+            }}
+          >
+            <div>
+              <H2Icon />
+            </div>
+            <div>
+              <p className='text-sm'>Heading 2</p>
+            </div>
+          </button>
```

Comment:
whenever there is some repetitive code, try extract that to separate component, to make code clean and maintainable